docs/deploy.py

```diff
@@ -32,38 +31,42 @@ def ensure_not_exists(path):
 
 
 def main():
+    old_dir = os.getcwd()
     print("Moving to %s." % HERE)
-    with path(HERE):
-        print("Building docs with 'make html'")
-        check_call(['make', 'html'])
-
-        print("Clearing temp location '%s'" % TEMP_LOCATION)
-        rmtree(TEMP_LOCATION, ignore_errors=True)
-
-        with removing(TEMP_LOCATION):
-            print("Copying built files to temp location.")
-            move('build/html', TEMP_LOCATION)
-
-            print("Moving to '%s'" % ZIPLINE_ROOT)
-            os.chdir(ZIPLINE_ROOT)
-
-            print("Checking out gh-pages branch.")
-            check_call(
-                [
-                    'git', 'branch', '-f',
-                    '--track', 'gh-pages', 'origin/gh-pages'
-                ]
-            )
-            check_call(['git', 'checkout', 'gh-pages'])
-            check_call(['git', 'reset', '--hard', 'origin/gh-pages'])
-
-            print("Copying built files:")
-            for file_ in glob(TEMP_LOCATION_GLOB):
-                base = basename(file_)
-
-                print("%s -> %s" % (file_, base))
-                ensure_not_exists(base)
-                move(file_, '.')
+    os.chdir(HERE)
+
+    print("Building docs with 'make html'")
+    check_call(['make', 'html'])
+
+    print("Clearing temp location '%s'" % TEMP_LOCATION)
+    rmtree(TEMP_LOCATION, ignore_errors=True)
+
+    with removing(TEMP_LOCATION):
+        print("Copying built files to temp location.")
+        move('build/html', TEMP_LOCATION)
+
+        print("Moving to '%s'" % ZIPLINE_ROOT)
+        os.chdir(ZIPLINE_ROOT)
+
+        print("Checking out gh-pages branch.")
+        check_call(
+            [
+                'git', 'branch', '-f',
+                '--track', 'gh-pages', 'origin/gh-pages'
+            ]
+        )
+        check_call(['git', 'checkout', 'gh-pages'])
+        check_call(['git', 'reset', '--hard', 'origin/gh-pages'])
+
+        print("Copying built files:")
+        for file_ in glob(TEMP_LOCATION_GLOB):
+            base = basename(file_)
+
+            print("%s -> %s" % (file_, base))
+            ensure_not_exists(base)
+            move(file_, '.')
+
+    os.chdir(old_dir)
```

Comment:
Yup, so if we're inlining the `Path` context manager here, then what was a call to its `__exit__` method should end up in a `finally` block as you pasted, e.g.

```python
old_dir = self.getcwd()
os.chdir(HERE)  # was __enter__()
try:
    print("Building docs with 'make html'")
    check_call(['make', 'html'])
    ...
finally:
    os.chdir(old_dir)  # was __exit__()
```

As it is in the diff currently, an exception raised in `BLOCK`, e.g. in one of the `check_call`s, will not change the directory back to the original, whereas `with Path(HERE):` does.